### PR TITLE
[MLIR] Add attributes no_unwind and will_return to the LLVMIR dialect

### PR DESCRIPTION
**Pull request by FMarno**
And testing.
These are being added to be used in the GPU to LLVM SPV pass.

### CODE REVIEW

**Review by victor-eds — APPROVED** [review]

**Review by zero9178 — APPROVED** [review]

**Review by gysit — APPROVED** [review]
Thanks LGTM!

File: mlir/lib/Target/LLVMIR/ModuleTranslation.cpp [hunk 1]
```diff
@@ -1438,6 +1438,10 @@ static void convertFunctionAttributes(LLVMFuncOp func,
     llvmFunc->addFnAttr(llvm::Attribute::OptimizeNone);
   if (func.getConvergentAttr())
     llvmFunc->addFnAttr(llvm::Attribute::Convergent);
+  if (func.getNoUnwind())
+    llvmFunc->addFnAttr(llvm::Attribute::NoUnwind);
+  if (func.getWillReturn())
+    llvmFunc->addFnAttr(llvm::Attribute::WillReturn);
```
Comment (victor-eds) [inline mlir/lib/Target/LLVMIR/ModuleTranslation.cpp:1444]:
```suggestion
  if (func.getNoUnwindAttr())
    llvmFunc->addFnAttr(llvm::Attribute::NoUnwind);
  if (func.getWillReturnAttr())
    llvmFunc->addFnAttr(llvm::Attribute::WillReturn);
```
Minor for consistency with other attributes. Equivalent. `getNoUnwind` calls `getNoUnwindAttr`.

Comment (FMarno) [reply mlir/lib/Target/LLVMIR/ModuleTranslation.cpp]: good spot! I've updated that.

**Comment by victor-eds** [comment]
`buildkite/github-pull-requests` is blocked for this and other PRs. Merging this now as local tests pass fine.

**Pull request merged**